qutip/qip/device/processor.py

```diff
@@ -658,7 +664,9 @@ def run_analytically(self, init_state=None, qc=None):
 
         try:  # correct_global_phase are defined for ModelProcessor
             if self.correct_global_phase and self.global_phase != 0:
-                U_list.append(globalphase(self.global_phase, N=self.N))
+                U_list.append(globalphase(
+                    self.global_phase, N=self.num_qubits)
+                )
         except AttributeError:
```

Comment:
Not directly related to this PR, but what is the `AttributeError` that can be raised here?

Reply:
`global_phase` is only defined in some subclass but not in the parent class, like this one. It is only calculated if the pulses are compiled from gates manually. There is no global phase returned if one uses the optimal control algorithms for example. So it not always defined. But I think we can just define it to be 0 so that there is no need for error handling.

Reply:
Thanks! Let's leave that for another PR.